test/distributed/fsdp/test_fsdp_fine_tune.py

```diff
@@ -0,0 +1,163 @@
+# Owner(s): ["oncall: distributed"]
+
+import copy
+import sys
+from unittest import mock
+
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from torch.distributed.fsdp.fully_sharded_data_parallel import (
+    FullyShardedDataParallel as FSDP,
+    ShardingStrategy,
+)
+from torch.distributed.fsdp.wrap import ModuleWrapPolicy
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest
+from torch.testing._internal.common_utils import run_tests, TEST_WITH_DEV_DBG_ASAN
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class TestFSDPFineTune(FSDPTest):
+    """Tests fine-tuning cases where some parameters are frozen."""
+
+    NUM_LINEARS = 6
+
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    def _init_seq_module(self) -> nn.Module:
+        torch.manual_seed(42)
```

Comment:
should we have a different seed for each rank so grads on each rank are different?

Reply:
We should use the same seed for model initialization, and then on L148 I have:
```
torch.manual_seed(self.rank + 1)
```